screens/Budget/BudgetScreen.tsx

```diff
@@ -41,32 +41,30 @@ function HeaderButton({ direction, onPress, marginLeft, marginRight }: HeaderBut
 }
 
 export default function BudgetScreen({ navigation, route }: RootTabScreenProps<'Budget'>) {
-    const passwordHash = useAuth();
     const now = moment();
     const [month, setMonth] = useState(MONTHS_ORDER[now.month()]);
     const [year, setYear] = useState(now.year());
 
-    const { data: budgetData, refetch: budgetRefetch } = useQuery<GetBudgetsQuery, GetBudgetsQueryVariables>(GetBudgetsDocument, {
-        variables: { passwordHash },
+    const [getBudgets, { data: budgetData, refetch: budgetRefetch }] = useLazyQuery<GetBudgetsQuery, GetBudgetsQueryVariables>(GetBudgetsDocument);
+    const [getMonthlyBreakdown, { data: monthData, refetch: monthRefetch }] = useLazyQuery<GetMonthBreakdownQuery, GetMonthBreakdownQueryVariables>(GetMonthBreakdownDocument);
+    const passwordHash = useAuth({
+        onRetrieved: (passwordHash) => {
+            getBudgets({ variables: { passwordHash } });
+            getMonthlyBreakdown({ variables: { passwordHash, month: month as MonthType, year } });
+        },
+        redirect: 'ifUnauthorized',
     });
-    const { data: monthData, refetch: monthRefetch } = useQuery<GetMonthBreakdownQuery>(GetMonthBreakdownDocument, {
-        variables: {
-            passwordHash,
-            month,
-            year
-        }
+    useRefresh(() => {
+        budgetRefetch();
+        monthRefetch();
```

Comment:
You don't need to fetch here, it will do the fetch when the passHash resolves.

Reply:
useRefresh is for when data is changed and user comes back to this page. Eg, after editting a budget category.

Reply:
oh shoot you are right, I read that as useEffect.